File: app/src/main/java/me/shkschneider/app/AboutFragment.java
package me.shkschneider.app;

import android.os.Bundle;
import android.preference.Preference;

import me.shkschneider.skeleton.AndroidHelper;
import me.shkschneider.skeleton.ApplicationHelper;
import me.shkschneider.skeleton.SkeletonPreferenceFragment;

public class AboutFragment extends SkeletonPreferenceFragment {

    @Override
    public void onCreate(final Bundle paramBundle) {
        super.onCreate(paramBundle);
        addPreferencesFromResource(R.xml.fragment_about);

        // Application

        final Preference appPackage = findPreference("app_package");
        appPackage.setTitle("packageName");
        appPackage.setSummary(ApplicationHelper.packageName());

        final Preference appVersionName = findPreference("app_versionName");
        appVersionName.setTitle("versionName");
        appVersionName.setSummary(ApplicationHelper.versionName());

        final Preference appVersionCode = findPreference("app_versionCode");
        appVersionCode.setTitle("versionCode");
        appVersionCode.setSummary(String.valueOf(ApplicationHelper.versionCode()));

        // OS

        final Preference osVersion = findPreference("os_version");
        osVersion.setTitle("version");
        osVersion.setSummary(AndroidHelper.versionName());

        final Preference osApi = findPreference("os_api");
        osApi.setTitle("api");
        osApi.setSummary(String.valueOf(AndroidHelper.api()));
    }

}

File: demo/src/main/kotlin/me/shkschneider/skeleton/demo/about/AboutFragment.kt
package me.shkschneider.skeleton.demo.about

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import me.shkschneider.skeleton.android.BuildConfig
import me.shkschneider.skeleton.android.app.ApplicationHelper
import me.shkschneider.skeleton.android.os.AndroidHelper
import me.shkschneider.skeleton.android.util.Metrics
import me.shkschneider.skeleton.demo.R
import java.util.Locale

class AboutFragment : PreferenceFragmentCompat() {

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.about, rootKey)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = super.onCreateView(inflater, container, savedInstanceState)
        // If preferences have no icons, this is needed to get rid of left's empty space
        // In Material (preference_material.xml) PreferenceCompat (android.R.id.icon): minWidth=60dp
        listView?.setPadding(-1 * Metrics.pixelsFromDp(60.toFloat()), 0, 0, 0)
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        // Application
        findPreference<Preference>("app_package")?.run {
            title = "PackageName"
            summary = ApplicationHelper.packageName
        }
        findPreference<Preference>("app_versionName")?.run {
            title = "VersionName"
            summary = ApplicationHelper.versionName()
        }
        findPreference<Preference>("app_versionCode")?.run {
            title = "VersionCode"
            summary = ApplicationHelper.versionCode().toString()
        }
        findPreference<Preference>("app_revision")?.run {
            title = "Revision"
            summary =  BuildConfig.REVISION
        }
        findPreference<Preference>("app_variant")?.run {
            title = "Variant"
            summary = (if (ApplicationHelper.debuggable) "debug" else "release").toUpperCase(Locale.getDefault())
        }
        // OS
        findPreference<Preference>("os_version")?.run {
            title = "Version"
            summary = AndroidHelper.codename()
        }
        findPreference<Preference>("os_api")?.run {
            title = "API"
            summary = AndroidHelper.api().toString()
        }
    }

}
